patternfly/patternfly-elements
# Cpfed 3805 set focus on pfe nav close

---
name: CPFED 3805 Set focus when closing nav with escape key
labels: "feature, ready for review"
---

## pfe-navigation

- code cleanup and comment clean up
- update keydown event listener to use logic to close menu even when mobile menu dropdown is open
- remove unneeded mobile menu keydown event listener
- add removeEventListeners to disconnected callback
- add dynamic settings for aria controls, aria hidden, aria expanded for util buttons and mobile menu toggle
- update general keyboard listener function to use event.key instead of event.which bc that is deprecated
- added note to look into bug on firefox: escape key keydown function only fires if you have tabbed into the menu but not if you have just clicked the menu open with the mouse click

### Testing instructions

1. Pull down this branch
2. Start up pfe-nav demo locally
3. Tab through the menu items on the /single-nav.html demo page and press the escape key, this should set the focus on the trigger button and close the entire nav


### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.
